modules/xmpp/XmppConnection.js

```diff
@@ -260,7 +264,9 @@ export default class XmppConnection extends Listenable {
 
         if (status === Strophe.Status.CONNECTED || status === Strophe.Status.ATTACHED) {
             this._maybeEnableStreamResume();
-            this._maybeStartWSKeepAlive();
+
+            // after connecting immediately check whether shard changed
+            this._maybeStartWSKeepAlive(this._options, 0);
```

Comment:
if you need to check shard on the demand it's better extract the fetch and verification part into a separate function and call it here instead of calling "maybe start keep alive" with the intention of "do the fetch request and check if shard changed"